### PR TITLE
Fix Y error labels in muon analysis results tables

### LINKED ISSUE #29409: Muon analysis results table
In results table the output no longer has the y errors set in the table columns

https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py#L276

it might be that the value has changed at
https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py#L33

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**

The output results table from Muon analysis should label the colombs with `Y0, Y1, Y2...` and `Y0_err, Y1_err, Y2_err...` for the values and errors respectivly. This pr links the two colombs so this label is propperly added to the errors.

**To test:**

1. Open Muon Analysis.
2. Load EMU20918.
3. Perform a fit.
4. Go to the results tab and output the results.
5. check that the colomns in the results table have the right labels linking values with their errors.

Fixes #29409 

*This does not require release notes* because it fixes a bug that was not in the previous release.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
Functionality looks good. We do need to update the release notes and I think we should also update https://docs.mantidproject.org/nightly/concepts/TableWorkspaces.html to show the new functionality. What do you think @Harrietbrown ?

File: scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py [hunk 1]
```diff
@@ -275,6 +275,7 @@ def float_log(wksp_name, log_name):
             if _param_error_should_be_displayed(name):
                 table.addColumn('float', _error_column_name(name),
                                 TableColumnType.YErr.value)
+                table.setLinkedYCol(table.columnCount()-1, table.columnCount()-2)
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py:278]: can we have something to explain why -1 and -2?

**Review by AnthonyLim23 — DISMISSED** [review]
Documentation looks good and it works as expected.

File: Framework/PythonInterface/mantid/api/src/Exports/ITableWorkspace.cpp [hunk 2]
```diff
@@ -657,8 +676,11 @@ void export_ITableWorkspace() {
               "Label)."))
 
       .def("getLinkedYCol", &getLinkedYCol, (arg("self"), arg("column")),
-           "Get the plot type of given column as an integer. "
-           "Accepts column name or index. ")
+           "set the data column associated with a given error column. ")
```
Comment (martyngigg) [inline Framework/PythonInterface/mantid/api/src/Exports/ITableWorkspace.cpp:679]: Nice spot that we were missing the set method exposed. Should this say **Get** here?

File: Framework/PythonInterface/mantid/api/src/Exports/ITableWorkspace.cpp [hunk 3]
```diff
@@ -657,8 +676,11 @@ void export_ITableWorkspace() {
...
+
+      .def("setLinkedYCol", &setLinkedYCol,
+           (arg("self"), arg("errColumn"), arg("dataColumn")),
+           "set the data column associated with a given error column. ")
```
Comment (martyngigg) [inline Framework/PythonInterface/mantid/api/src/Exports/ITableWorkspace.cpp:683]: Could we capitalize the `Set` for consistency with the other methods?

**Review by martyngigg — CHANGES_REQUESTED** [review]

**Review by martyngigg — APPROVED** [review]
Thanks for making those changes.

**Comment by martyngigg** [comment]
Unrelated test failure.

**Pull request merged**